server/src/main/java/com/flipkart/varadhi/verticles/webserver/TrafficAggregator.java

```diff
@@ -0,0 +1,99 @@
+package com.flipkart.varadhi.verticles.webserver;
+
+import com.flipkart.varadhi.cluster.MessageExchange;
+import com.flipkart.varadhi.cluster.messages.ClusterMessage;
+import com.flipkart.varadhi.qos.client.RateLimiterService;
+import com.flipkart.varadhi.qos.entity.LoadInfo;
+import com.flipkart.varadhi.qos.entity.SuppressionData;
+import com.flipkart.varadhi.qos.entity.SuppressionFactor;
+import com.flipkart.varadhi.qos.entity.TrafficData;
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+import static com.flipkart.varadhi.core.cluster.ControllerApi.ROUTE_CONTROLLER;
+
+/**
+ * This class will capture incoming traffic usage by providing a method to add topic usage by producers.
+ * This class will accumualte everything and send to the controller for further processing for rate limit.
+ * Post that, reset the usage for the topic.
+ */
+@Slf4j
+public class TrafficAggregator {
```

Comment:
so as discussed, this aggregator and send class, is relying on variable time width. the different between "to" and "from" wont always be same.

Reply:
This is taken care of. While calculating the suppression factor, accurate `from` and `to` are used.